arm9/source/patches/platform/datel/DATELSpiCommandsAsm.h

```diff
@@ -0,0 +1,92 @@
+#pragma once
+#include "sections.h"
+#include "../SdReadDmaPatchCode.h"
+
+DEFINE_SECTION_SYMBOLS(datel_ntr_command);
+DEFINE_SECTION_SYMBOLS(datel_read_spi);
+DEFINE_SECTION_SYMBOLS(datel_spi_send);
+
+extern "C" void DATEL_CycleSpi();
+
+extern "C" u8 DATEL_ReadSpiByte();
+extern "C" u8 DATEL_ReadWriteSpiByte(u8 value);
+extern "C" u8 DATEL_ReadSpiByteTimeout();
+extern "C" bool DATEL_WaitSpiByteTimeout();
+
+extern "C" u8 DATEL_SpiSendSDIOCommandR0(u32 arg, u8 cmd);
+extern "C" u8 DATEL_SpiSendSDIOCommand(u32 arg, u8 cmd, int extraBytes);
+
+extern u32 DATEL_CycleSpi_ReadSpiByte;
+
+extern u32 DATEL_SpiSendSDIOCommandR0_CycleSpi;
+
+extern u32 DATEL_SpiSendSDIOCommandR0_ReadWriteSpiByte;
+extern u32 DATEL_SpiSendSDIOCommandR0_ReadSpiByteTimeout;
+
+class DATELReadSpiBytePatchCode : public PatchCode
+{
+public:
+    explicit DATELReadSpiBytePatchCode(PatchHeap& patchHeap)
+        : PatchCode(SECTION_START(datel_read_spi), SECTION_SIZE(datel_read_spi), patchHeap) { }
+
+    const void* GetReadSpiByteFunction() const
+    {
+        return GetAddressAtTarget((void*)DATEL_ReadSpiByte);
+    }
+
+    const void* GetReadWriteSpiByteFunction() const
+    {
+        return GetAddressAtTarget((void*)DATEL_ReadWriteSpiByte);
+    }
+
+    const void* GetWaitSpiByteTimeoutFunction() const
+    {
+        return GetAddressAtTarget((void*)DATEL_WaitSpiByteTimeout);
+    }
+
+    const void* GetReadSpiByteTimeoutFunction() const
+    {
+        return GetAddressAtTarget((void*)DATEL_ReadSpiByteTimeout);
+    }
+};
+
+class DATELCycleSpiPatchCode : public PatchCode
+{
+public:
+    explicit DATELCycleSpiPatchCode(PatchHeap& patchHeap,
```

Comment:
```suggestion
    DATELCycleSpiPatchCode(PatchHeap& patchHeap,
```
`explicit` is only needed when the constructor has a single argument